### PR TITLE
RETprojects/niceIdeaCommentFeed

### PR DESCRIPTION
**Pull request by RETprojects**
I added the idea_card attribute to the statement in CommentForm's handleSubmit method and passed in selectedCard. Now new comments submitted by the user will be displayed in the workshopping page comment feed upon refreshing the page.

### CODE REVIEW

File: apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js [hunk 1]
```diff
@@ -65,10 +67,61 @@ function IdeaCard({ cards, cardType }) {
...
     >
+
       <atoms.Box>
-        <IdeaCardTag status={tagContent} />
+        <IdeaCardTag
+          status={tagContent}
+        />
       </atoms.Box>
 
+      <IdeaCardImg
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js:77]: this looks like it's duplicated with line 125

File: apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js [hunk 2]
```diff
@@ -65,10 +67,61 @@ function IdeaCard({ cards, cardType }) {
...
+          <atoms.Button
+            buttonSize='standard'
+            buttonType='alternative'
+            style={{ margin: '0rem 2rem 1.5rem' }}
+          >
+            {buttonContent}
+          </atoms.Button>
+        </Link>
+      )}
+      <UpdateFailure />
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js:123]: this looks like a duplicate with line 177

File: apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js [hunk 3]
```diff
@@ -65,10 +67,61 @@ function IdeaCard({ cards, cardType }) {
...
+          />
+
+          <IdeaCardUpdated
+            updatedAt={cards.updated_at}
+          />
+        </atoms.Box>) : null}
+        </a>
+      </Link>
+
+      {tagContent == "archived" ? (
```
Comment (dbradham) [inline apps/ideaspace/src/components/common/IdeaCard/IdeaCard.js:103]: this looks like a duplicate of line 157

File: packages/UI/src/context/UserDataContext.js [hunk 4]
```diff
@@ -2,13 +2,35 @@ import constate from 'constate'; // State Context Object Creator
 import { useEffect, useState } from 'react';
 import axios from "axios";
 
+// export const DEFAULT_USER = {
```
Comment (dbradham) [inline packages/UI/src/context/UserDataContext.js:5]: We need to keep the default user as is, otherwise users will see Kris' info when they are not logged in which will cause a lot of confusion.

File: packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx [hunk 5]
```diff
@@ -1,7 +1,9 @@
+import axios from 'axios';
 import Like from '../../../../assets/icons/Like';
 import Box from '../../../atoms/Box';
 import { InteractionButton } from '../StyledCommonComponents';
 import type { LikeButtonProps } from '.';
+import { useUserDataContext } from '@devlaunchers/components/context/UserDataContext';
```
Comment (dbradham) [inline packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx:6]: is this change needed?

File: packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx [hunk 6]
```diff
@@ -1,7 +1,9 @@
+import axios from 'axios';
```
Comment (dbradham) [inline packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx:1]: is this change needed?

File: apps/ideaspace/src/pages/workshop/[ideaId].js [hunk 7]
```diff
@@ -1,7 +1,28 @@
 import Head from 'next/head';
 import WorkshoppingPage from '../../components/modules/WorkshoppingPage/WorkshoppingPage';
+import React from 'react';
+import { useRouter } from "next/router";
+
+export default function WorkshopIdeaPage() {
+  const router = useRouter();
+  const { idea } = router.query;
```
Comment (dbradham) [inline apps/ideaspace/src/pages/workshop/[ideaId].js:8]: is the `router` and `idea` being used for anything here?

File: apps/ideaspace/src/pages/workshop/[ideaId].js [hunk 8]
```diff
@@ -1,7 +1,28 @@
...
+  //   return null;
+  // }
+
+  // // path is an array of path after ideaId
+  // const ideaId = idea[1];
+  // console.log(`square bracket file ideaId=${ideaId}`);
+  // return (
+  //   <div>
+  //     <Header>
+  //     <WorkshoppingPage ideaId={ideaId} />
```
Comment (dbradham) [inline apps/ideaspace/src/pages/workshop/[ideaId].js:21]: can we delete these commented lines of code?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/WorkshoppingPage.js [hunk 9]
```diff
@@ -92,4 +124,4 @@ function WorkshoppingPage() {
   }
 }
 
-export default WorkshoppingPage;
+// export default WorkshoppingPage;
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/WorkshoppingPage.js:127]: if we are adding the export statement to the same line as where we define the function, let's just delete this line altogether instead of commenting it out

File: apps/ideaspace/src/components/modules/WorkshoppingPage/WorkshoppingPage.js [hunk 10]
```diff
@@ -26,12 +27,34 @@ import {
...
+
   const router = useRouter()
-  const { ideaId } = router.query;
+  //console.log("Router: ", router)
+  //console.log("Router query: ", router.query)
+  if(!router.isReady){
+    console.log("It's NOT ready yet!")
+  }
+  else{
+    console.log("It's ready!")
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/WorkshoppingPage.js:40]: can we remove these `console.log` statements?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/SingleComment.js [hunk 11]
```diff
@@ -1,22 +1,62 @@
...
     <UserImage alt="user_image" src={`https://picsum.photos/70?random=${props.id}`} />
     <div className="textContent">
       <SingleCommentContent>
         <h3>{props.author}</h3>
-        <div source={props.children} ><p>{props.children}</p></div>
+        {/* get the idea ID from the URL if possible and determine the idea owner (maybe do this in another file) */}
+      </SingleCommentContent>
+      <SingleCommentContent>
+        {/* date of creation here, i.e. "2 days ago" */}
+        {/* <h5>{new Date(props.createdAt)[Symbol.toPrimitive]("string")}</h5> */}
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/SingleComment.js:51]: can we delete this line?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js [hunk 12]
```diff
@@ -1,16 +1,23 @@
...
+  // SubmitButton,
 } from './StyledComments.js';
 import { useUserDataContext } from '@devlaunchers/components/context/UserDataContext';
 import SignInButton from '../../../common/SignInButton/SignInButton';
 import { agent } from '@devlaunchers/utility';
 
 function CommentForm(props) {
+  const [state, setState] = React.useState(false)
+  // state = {
+  //   reload: false
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js:18]: can we delete these commented lines?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js [hunk 13]
```diff
@@ -27,29 +34,52 @@ function CommentForm(props) {
...
-    console.log('res', res);
-    props.setHandleTextChange('');
+    try {
+      const res = await agent.Comments.post(data);
+      console.log('data for new comment:', data)
+      console.log('res:', res);
+      props.setHandleTextChange('');
+
+      // submit the new comment to the comment feed
+      console.log("comments after new comment:", props.comments)
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js:50]: can we remove the `console.log`s in this method?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js [hunk 14]
```diff
@@ -27,29 +34,52 @@ function CommentForm(props) {
...
-        data
-      )
-      .then((response) => {
-        if (response.status === 200) {
-          props.setHandleTextChange('');
-        }
-      });
+    // axios
+    //   .post(
+    //     `${process.env.NEXT_PUBLIC_STRAPI_URL}/api/idea-cards/${selectedCard.id}/comment`,
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/CommentForm.js:57]: can we remove this commented code?

File: apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/DisplayComments.js [hunk 15]
```diff
@@ -2,35 +2,78 @@ import React, { useEffect, useState } from 'react';
...
 
-  useEffect(async () => {
-    if (props.selectedCard.id != undefined) {
+  console.log("props: ", props)
+
+  // useEffect(async () => {
+  //   // if (props.selectedCard.id != undefined) {
+  //   //   axios.get(`${process.env.NEXT_PUBLIC_STRAPI_URL}/idea-cards/${props.selectedCard.id}`)
+  //   //      .then(response => { // call backend
+  //   //     console.log(response.data)
```
Comment (dbradham) [inline apps/ideaspace/src/components/modules/WorkshoppingPage/CommentsForm/DisplayComments.js:30]: can we remove this commented code?

File: packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx (hunk 5 above)
Comment (RETprojects) [reply packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx:6]: it doesn't look like it's being read

File: apps/ideaspace/src/pages/workshop/[ideaId].js (hunk 7 above)
Comment (RETprojects) [reply apps/ideaspace/src/pages/workshop/[ideaId].js:8]: they aren't being used

Comment (dbradham) [reply apps/ideaspace/src/pages/workshop/[ideaId].js:8]: Can we remove it then?

File: packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx (hunk 5 above)
Comment (dbradham) [reply packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx:6]: Can we remove it?

File: apps/ideaspace/src/pages/workshop/[ideaId].js (hunk 7 above)
Comment (RETprojects) [reply apps/ideaspace/src/pages/workshop/[ideaId].js:8]: I removed it, a commit is coming soon

File: packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx (hunk 5 above)
Comment (RETprojects) [reply packages/UI/src/components/molecules/InteractionButtons/LikeButton/LikeButton.tsx:6]: I removed it, a commit is coming soon

**Review by dbradham — APPROVED** [review]
Looks good

**Comment by dbradham** [comment]
@dev-launchers/platform-enablement please take a look at the changes here when you are able. Thank you!

File: packages/utility/agent.ts [hunk 16]
```diff
@@ -138,6 +138,7 @@ const User = {
 
 const Comments = {
   put: (id: string, body: {}) => requests.put<Comment>(id, body),
+  post: (body: {}) => requests.post<Comment>("comments", body),
```
Comment (Enjoy2Live) [inline packages/utility/agent.ts:141]: why the body is typed as `{}`? I think it should be `(body: Comment) => requests.post<Comment>("comments", body)`

File: packages/UI/src/context/UserDataContext.js [hunk 17]
```diff
@@ -23,6 +23,61 @@ export const DEFAULT_USER = {
...
+//       id: 4,
+//       published_at: '2021-08-11T05:36:30.868Z',
+//       created_at: '2021-08-11T05:36:25.579Z',
+//       updated_at: '2021-08-11T05:36:31.097Z',
+//       interest: 'UI/UX',
+//       categories: [],
+//     },
+//   ],
+// };
+
```
Comment (Enjoy2Live) [inline packages/UI/src/context/UserDataContext.js:80]: why do we have this huge commented chunk? can we remove it if not needed?

**Review by Enjoy2Live — CHANGES_REQUESTED** [review]
Please go over the requested changes and let me know of any questions you may have here. 

it's great to see meaningful contributions to the `agent.ts` :smile:

**Comment by dbradham** [comment]
> Please go over the requested changes and let me know of any questions you may have here.
> 
> it's great to see meaningful contributions to the `agent.ts` 😄

Thanks Mohammed! It looks like Remi has implemented the changes. Can you let him know if there is anything else to change?

**Comment by dbradham** [comment]
> > Please go over the requested changes and let me know of any questions you may have here.
> > it's great to see meaningful contributions to the `agent.ts` 😄
> 
> Thanks Mohammed! It looks like Remi has implemented the changes. Can you let him know if there is anything else to change?

@Enjoy2Live friendly reminder here.

**Review by Enjoy2Live — APPROVED** [review]
